### ISSUE TITLE
"? ... : null" is not inferred as a nullable type

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#39437]
Take this program:

```dart
bool b = false;
int i = b ? 123 : null;

int test() {
  return b ? 123 : null;
}
```

And analyze it with:

```sh
dartanalyzer --enable-experiment=non-nullable temp.dart
```

I would expect two errors, because the result of `?:` is nullable and it is being used in non-nullable contexts. Instead, I get:

```
No issues found!
```

**Comment by stereotype441** [comment]
It looks like the analyzer's current implementation of LUB thinks that the LUB of types `int` and `Null` is `int` (which was correct pre-NNBD but is wrong under the new NNBD rules).

CC @scheglov who is currently working on improvements to LUB

**Comment by stereotype441** [comment]
@scheglov confirmed that under the new rules he's implementing, `LUB(int, Null)` will be `int?`, so this should work fine once that work is done.

**Comment by scheglov** [comment]
Fixed in 6444fba9c71d5125799afe5798afb6f26a110b43.
![image](https://user-images.githubusercontent.com/384794/69375360-ff6bc980-0c5c-11ea-83e6-bfe0b849e004.png)

**Comment by munificent** [comment]
Thanks!
